File: t/CS/ImageProcessingDemo/UI/Command/ChangeViewPerspectiveDialog.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Leadtools;
using Leadtools.ImageProcessing;

namespace ImageProcessingDemo
{
   //This dialog is used to allow the user to specify values which
   //will be used for the ChangeViewPerspectiveCommand

   public partial class ChangeViewPerspectiveDialog : Form
   {
      private ChangeViewPerspectiveCommand _ChangeViewPerspectiveCommand;

      public ChangeViewPerspectiveDialog()
      {
         InitializeComponent();
         _ChangeViewPerspectiveCommand = new ChangeViewPerspectiveCommand();

         //Set command default values
         InitializeUI();

      }
      private void InitializeUI()
      {
         string[] names;
         Array numbers;

         names = Enum.GetNames(typeof(RasterViewPerspective));
         numbers = Enum.GetValues(typeof(RasterViewPerspective));

         for (int i = 1; i <= 8; i++)
         {
            _cmbViewPerspective.Items.Add(((RasterViewPerspective)i).ToString());
         }

         _cmbViewPerspective.SelectedIndex = 0;
      }
      public ChangeViewPerspectiveCommand ChangeViewPerspectivecommand
      {
         get
         {
            //Update command values
            UpdateCommand();
            return _ChangeViewPerspectiveCommand;
         }
         set
         {
            _ChangeViewPerspectiveCommand = value;
            InitializeUI();
         }
      }
      private void UpdateCommand()
      {
         _ChangeViewPerspectiveCommand.ViewPerspective = TranslateViewPerspective();
      }

      public RasterViewPerspective TranslateViewPerspective()
      {
         switch (_cmbViewPerspective.SelectedItem.ToString())
         {
            case "BottomLeft":
               return RasterViewPerspective.BottomLeft;
            case "BottomLeft180":
               return RasterViewPerspective.BottomLeft180;
            case "BottomLeft270":
               return RasterViewPerspective.BottomLeft270;
            case "BottomLeft90":
               return RasterViewPerspective.BottomLeft90;
            case "BottomRight":
               return RasterViewPerspective.BottomRight;
            case "LeftBottom":
               return RasterViewPerspective.LeftBottom;
            case "LeftTop":
               return RasterViewPerspective.LeftTop;
            case "RightBottom":
               return RasterViewPerspective.RightBottom;
            case "RightTop":
               return RasterViewPerspective.RightTop;
            case "TopLeft":
               return RasterViewPerspective.TopLeft;
            case "TopLeft180":
               return RasterViewPerspective.TopLeft180;
            case "TopLeft270":
               return RasterViewPerspective.TopLeft270;
            case "TopLeft90":
               return RasterViewPerspective.TopLeft90;
            case "TopRight":
               return RasterViewPerspective.TopRight;
            default:
               return RasterViewPerspective.TopLeft;
         }
      }

      private void _btnok_Click(object sender, EventArgs e)
      {
         UpdateCommand();
         this.DialogResult = DialogResult.OK;
         this.Close();
      }

      private void _btncancel_Click(object sender, EventArgs e)
      {
         this.DialogResult = DialogResult.Cancel;
      }
   }
}

File: t/VB/ImageProcessingDemo/UI/Command/ChangeViewPerspectiveDialog.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports Leadtools
Imports Leadtools.ImageProcessing

Namespace ImageProcessingDemo
   'This dialog is used to allow the user to specify values which
   'will be used for the ChangeViewPerspectiveCommand

   Partial Public Class ChangeViewPerspectiveDialog : Inherits Form
      Private _ChangeViewPerspectiveCommand As ChangeViewPerspectiveCommand

      Public Sub New()
         InitializeComponent()
         _ChangeViewPerspectiveCommand = New ChangeViewPerspectiveCommand()

         'Set command default values
         InitializeUI()

      End Sub
      Private Sub InitializeUI()
            Dim names As String()
            Dim Values As Array
            names = System.Enum.GetNames(GetType(RasterViewPerspective))
            Values = System.Enum.GetValues(GetType(RasterViewPerspective))
            Dim i As Integer

            For i = 1 To 8
                _cmbViewPerspective.Items.Add(CType(i, RasterViewPerspective).ToString())
            Next

            _cmbViewPerspective.SelectedIndex = 0
        End Sub
      Public Property ChangeViewPerspectivecommand() As ChangeViewPerspectiveCommand
         Get
            'Update command values
            UpdateCommand()
            Return _ChangeViewPerspectiveCommand
         End Get
         Set(ByVal value As ChangeViewPerspectiveCommand)
            _ChangeViewPerspectiveCommand = value
            InitializeUI()
         End Set
      End Property
      Private Sub UpdateCommand()
         _ChangeViewPerspectiveCommand.ViewPerspective = TranslateViewPerspective()
      End Sub

      Public Function TranslateViewPerspective() As RasterViewPerspective
         Select Case _cmbViewPerspective.SelectedItem.ToString()
            Case "BottomLeft"
               Return RasterViewPerspective.BottomLeft
            Case "BottomLeft180"
               Return RasterViewPerspective.BottomLeft180
            Case "BottomLeft270"
               Return RasterViewPerspective.BottomLeft270
            Case "BottomLeft90"
               Return RasterViewPerspective.BottomLeft90
            Case "BottomRight"
               Return RasterViewPerspective.BottomRight
            Case "LeftBottom"
               Return RasterViewPerspective.LeftBottom
            Case "LeftTop"
               Return RasterViewPerspective.LeftTop
            Case "RightBottom"
               Return RasterViewPerspective.RightBottom
            Case "RightTop"
               Return RasterViewPerspective.RightTop
            Case "TopLeft"
               Return RasterViewPerspective.TopLeft
            Case "TopLeft180"
               Return RasterViewPerspective.TopLeft180
            Case "TopLeft270"
               Return RasterViewPerspective.TopLeft270
            Case "TopLeft90"
               Return RasterViewPerspective.TopLeft90
            Case "TopRight"
               Return RasterViewPerspective.TopRight
            Case Else
               Return RasterViewPerspective.TopLeft
         End Select
      End Function

      Private Sub _btnok_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btnok.Click
         UpdateCommand()
         Me.DialogResult = DialogResult.OK
         Me.Close()
      End Sub

      Private Sub _btncancel_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btncancel.Click
         Me.DialogResult = DialogResult.Cancel
      End Sub
   End Class
End Namespace
